### DISCUSSION TITLE
Prevent passing a function reference directly to iterator methods

### DISCUSSION THREAD

**Discussion by user72356** [eslint/eslint#16119] [Q&A] [answered]
Is there an eslint rule to stop function references being used as callbacks without their parameters being explicitly given? For instance I want to avoid a situation like this:

```js
export function toReadableNumber(num) {
  // Return num as string in a human readable form.
  // Eg 10000000 might become '10,000,000'
}
```
```js
// Convert some numbers into human-readable strings:
import { toReadableNumber } from 'some-library';
const readableNumbers = someNumbers.map(toReadableNumber); // toReadableNumber passed as a function ref with the argument being implicit
```

Instead I want to force this:

```js
// Convert some numbers into human-readable strings:
import { toReadableNumber } from 'some-library';
const readableNumbers = someNumbers.map(num => toReadableNumber(num)); // toReadableNumber is passed as an arrow function with the parameter being explicit.
```

For more details on the problem I'm trying to avoid see this article: https://jakearchibald.com/2021/function-callback-risks/

`prefer-arrow-callback` doesn't seem to do the trick. Interestingly I found this https://github.com/sindresorhus/eslint-plugin-unicorn that does what I need. I'm wondering if there is a native built-in eslint rule for that.

**Comment by mdjermanovic** [comment, marked as answer]
There is no built-in rule for this. You can use [unicorn/no-array-callback-reference](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md) plugin rule.

**Comment by librz** [comment]
I came across a bug one time with the `parseInt` function:

```js
const arr = ["1", "1.2", "3.5"];
const nums = arr.map(parseInt);
console.log(nums);
```

This code will not show any warning even in TypeScript environment.

`parseInt` can take a 2nd (optional) argument as radix, array callback provides the 2nd argument but is meant for index, hence the unexpected behavior.

I wish the `no-array-callback-reference` rule could be included in the official eslint, and allows it if the function reference passed in is guaranteed to only take ONE argument.
